source/core/TypeTemplates.cpp

```diff
@@ -147,13 +147,14 @@ void TypeTemplateVisitor::VisitCluster(ClusterType* type)
     AggregateAlignmentCalculator* saveCalc = _alignmentCalculator;
     _alignmentCalculator = &calc;
 
-    TypeRef elementTypes[1000];   // TODO(PaulAustin): enforce limits or make them dynamic
     IntIndex subElementCount = type->SubElementCount();
+    TypeRef* elementTypes = new TypeRef[subElementCount];
```

Comment:
As discussed in person last week, why don't we make this a vector (as done above)?